drivers/soundwire/intel_init.c

```diff
@@ -56,26 +98,23 @@ static bool is_link_enabled(struct fwnode_handle *fw_node, int i)
 
 static int sdw_intel_cleanup(struct sdw_intel_ctx *ctx)
 {
-	struct sdw_intel_link_res *link = ctx->links;
+	struct sdw_intel_link_dev *ldev;
 	u32 link_mask;
 	int i;
 
-	if (!link)
-		return 0;
-
 	link_mask = ctx->link_mask;
 
-	for (i = 0; i < ctx->count; i++, link++) {
+	for (i = 0; i < ctx->count; i++) {
 		if (!(link_mask & BIT(i)))
 			continue;
```

Comment:
by construction below when a bit isn't set in that mask, the object is also not allocated, i.e. where `link_mask & BIT(i) == 0` there `ctx->ldev[i] == NULL`. So, this code is correct, but it makes you take the effort to stop and check that it's really correct :-) So, maybe at least a comment would be good. Or if we wanted to be somewhat defensive:
```
ldev = ctx->ldev[i];
if (!ldev) {
	if (link_mask & BIT(i))
		BIG_FAT_ERROR();
	continue;
}
```

Reply:
I added a comment but since we use the same pattern it's not really possible to reach a case with ldev==NULL